chess/4-database/database.md

```diff
@@ -114,19 +114,70 @@ As part of your unit test deliverable you need to meet the following requirement
 
 For this phase the TAs will grade the quality of your project's source code. The rubric used to evaluate code quality can be found here: [Rubric](../code-quality-rubric.md)
 
+### Common Submission Issues
+
+Since you're adding code that acts with an external program in Phase 4, the likelihood of having auto-grader test
+output different from your local output is higher than previous phases.
+
+If you encounter this, here are some common things to try (expand the collapsible section).
+
+<details>
```

Comment:
I have been trying to avoid using HTML syntax in MarkDown since it is not widely supported and would be complex to automatically transform for a different rendering.

Reply:
There are really two core issues here, and I want to address them each individually:

## Markdown Html Support
I don't know what you're talking about when you say that "HTML syntax in MarkDown is not widely supported."

All of the implementations I've used of MarkDown support standard HTML tags by treating MarkDown as a _superset_ of HTML. One of the problems with Markdown since the early days has been the inconsistency in implementations of the language, and this has been addressed and largely resolved by the CommonMark specification (see [Why is CommonMark needed?](https://commonmark.org/#:~:text=Why%20is%20CommonMark%20needed%3F)).

There are non-standard things that would expectably break on a migration to a different system, but using HTML isn't one of them. (Those would be GitHub specific features like: Alerts, code linking, and other niceties.) We are largely sticking to standard MarkDown and HTML practices, and we're doing good by not implementing our own homegrown version of a styling language; that has happened in other departments on campus, and it quickly becomes a limiting factor.

For our foreseeable purposes:
* Intelli-J renders HTML (and this particular element)
* GitHub renders HTML (and this particular element)
* Most common markdown interpreters render HTML

Are there any key platforms which notably _do not_ support HTML tags inside markdown?

## This particular `<details>` element
The HTML `<details>` element is a standard HTML tag labelled as "Baseline: widely available" by the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details). It is used to contain information in a collapsed-by-default state which users can then click to request additional information. This pattern is specifically documented and recommended by GitHub in an article titled [Organizing information with collapsed sections](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections). The pattern is also supported by [GitLab](https://docs.gitlab.com/ee/user/markdown.html#collapsible-section).

In this document, the `<details>` tag is used to wrap around a pretty beefy (and growing) section of debugging tips which common visitors to the page will not need. This improves the overall readability of the page by allowing students to reach the end of the document with the rubric and important notes without getting slogged down by the technical debugging details. Students who are currently facing an issue will be more than happy to open up the collapsed container and read about the solutions to their problems.

Alternative approaches to this organizational choice include:
* Leave the content flatly on the document and increase the overall length of the file
* Extract the debugging tips to a separate page where they can be isolated
* Remove the debugging tips and expect students to just figure it out

All considered, I feel like the collapsible section is a great blend of the best of each of these approaches. It gives students the information they need when they need it, but it avoid overburdening them with the details when they don't need it.

I do not feel like the use of HTML tags will hurt the portability of the project. The tag itself is a widely supported tag, and the feature is well documented, and multiple common flavors of markdown including the ones we use, support it.

## Screenshots
### Intelli-J
#### Collapsed (default)
<img width="1179" alt="image" src="https://github.com/user-attachments/assets/137eccfd-f110-4536-95b4-de3679a59850">

#### Expanded
<img width="1318" alt="image" src="https://github.com/user-attachments/assets/43341cf9-b363-473f-b076-ec812812bf53">


### GitHub
#### Collapsed (default)
<img width="1092" alt="image" src="https://github.com/user-attachments/assets/b5a86092-d53e-4258-ab57-03cd551abfed">

#### Expanded
<img width="1093" alt="image" src="https://github.com/user-attachments/assets/188e358e-28c1-46d0-a383-5cd6a2dafe3d">

Reply:
I moved the content into a separate page to account for the concerns about using HTML.